### DISCUSSION TITLE
Slower performance on Whisper model

### DISCUSSION THREAD

**Discussion by uanandaraja** [deepgram/community#269] [General help] [answered]
Hi!

I noticed a quite significant difference in performance, specifically Whisper Large model. Usually, a 1 hour audio will be done in 1-2 minutes, but now it's about 10-15 minutes. 

Thank you.

**Comment by jjmaldonis** [comment]
Whisper is currently having some issues. You can keep an eye on our [status](https://status.deepgram.com/) page to see when it's resolved.

**Reply by mosnicholas** [reply]
@jjmaldonis -- is it not updated in the status page?

**Reply by uanandaraja** [reply]
Yeah, I thought it was resolved? So the issue still persists yes?

**Reply by jjmaldonis** [reply]
I just learned that Whisper is currently experiencing an increase in latency. Spikes like this happen occasionally and they get resolved fairly quickly.

We have a couple engineers looking into whisper performance. If that yields results, you'll see some additional performance improvements.

**Reply by uanandaraja** [reply]
@jjmaldonis the latency is still quite bad as of today. Any updates?

**Reply by TomaszSzymanskiDl** [reply]
I recently tested Deepgram's Whisper Large model and OpenAI's hosting using a 1.7MB mp3 file from a Polish podcast. I used Deepgram free trial. The processing times were notably different, with Deepgram taking 23 seconds and OpenAI only 7 seconds.

The parameters used were:
   punctuate: true,
   smart_format: true,
   paragraphs: true

I'm curious - are your experiences similar? Is OpenAI's hosting typically faster, or could this be a one-off error? I'd appreciate your insights on this. Or maybe it is the problem of free Deepgram version I test.

**Reply by mosnicholas** [reply]
Yes @TomaszSzymanskiDl :(

**Comment by ramild** [comment]
Hey @jjmaldonis, we too have been noticing that Whisper has been slowing down quite a bit in the past month. A month ago, it used to process 1-2 minute audio files in about 10 seconds, which was a bit slower than OpenAI's Whisper at 6-7 seconds, but it was manageable.

Lately, for the same audio it may take between 40 seconds to 1.5 minutes, and sometimes even longer than 10 minutes. The response times have become unpredictable. Here are a couple of examples of request ids with extended wait times:

d83d7966-437b-4198-9423-d2e2d6571c0f (more than 2 minutes)
74bad510-8f97-437f-8164-d9c362263986 (more than 10 minutes)

Do you know when this might be fixed? Can we expect Whisper will be more stable and fast in the near future?

We're trying to figure out our game plan while we wait it to get fixed. Thanks!

**Comment by TomaszSzymanskiDl** [comment]
guys, did you notice any improvement in speed?

**Comment by jjmaldonis** [comment, marked as answer]
Hey all, early last week we improved the underlying systems that run Whisper to support the growing demand from our users. You should have seen significant improvements in the number of errors received and the latency of API requests to Whisper.

We will continue to improve the underlying infrastructure over the coming months, and may introduce additional rate limits if Whisper latency remains higher than desired.

Many of our users rely on Deepgram to serve voice-to-text solutions to their own customer base. When Whisper latency is high (and request take longer than normal to complete), those customers can see a poorer user experience because the audio files they upload seem to "hang" for an unknown amount of time. If necessary, additional rate limits will improve the end customer's experience by providing a near-instantaneous message saying, "this API request cannot be fulfilled in a reasonable amount of time", which allows products that rely on Deepgram to handle this scenario in a way that meets their customers' expectations. If you are using or plan to use thousands of dollars of Whisper requests per month, please keep an eye out for additional rate limits; you may also want to reach out to our sales team to continue the conversation.

**Reply by TomaszSzymanskiDl** [reply]
@jjmaldonis 
I regret to report that I'm still noticing significant slowness with the Whisper model. **A 6-minute file** processed using 'https://api.deepgram.com/v1/listen?model=general&tier=enhanced&language=pl&diarize=true&punctuate=true&smart_format=true&paragraphs=true' took just **10 seconds**, while the same file **took 2 minutes and 48 seconds when using Whisper** at 'https://api.deepgram.com/v1/listen?model=whisper-large&language=pl&diarize=true&punctuate=true&smart_format=true&paragraphs=true'.

We would be more than happy to use your proprietary model, but regrettably, the general/enhanced model seems to have a higher error rate in Polish language than the Whisper Large model.

What would you suggest? We're currently on a trial because we're considering different ASR solution providers.

**Reply by peldszus** [reply]
I also experienced much slower responses for whisper models than last time I tried. All around 5-7min response times for various input lengths 8-100min.

Since it doesn't seem to correlate with input length, I assume there is just more requests in total than the deployment can serve and we hang in a queue for a while?

@jjmaldonis What is the current rate limit for whisper models? Is it a customer specific rate limit or is applied to all customers? I ask because, the "pay-as-you-go"-plan speaks of 100 concurrent requests, but I just got a `{"err_code":"Too Many Requests","err_msg":"Please try again later." ...}` for my 3rd concurrent request.

**Reply by jjmaldonis** [reply]
Thanks for the replies. I'm keeping track of the feedback and we are continuing to watch the latency for Whisper. Please continue to post info about how Whisper is working for you in this thread (both good and bad results).

Here are some useful questions to answer:
* What is the worst case you have experienced?
* What is the typical case you experience?
* Do most calls complete quickly, and only a few complete slowly?
* What are the request IDs for the worst-case API calls?
* Are you being affected by rate limiting?

**Reply by jjmaldonis** [reply]
@peldszus each customer has an independent rate limit, and Whisper is rate limited to 5 concurrent requests unless you have a sales contract with us.

**Reply by ujlm** [reply]
Hi @jjmaldonis, could you please clarify what you mean exactly by a sales contract? Also, which plan does it correspond to?

I am experiencing degraded Whisper performance as well, with many requests timing out. The response I receive in such cases is:
`{
  err_code: 'Gateway Timeout',
  err_msg: 'Request timeout. Please try again later.',
  request_id: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
}
`

**Reply by jjmaldonis** [reply]
@ujlm a sales contract is a signed contract between you and Deepgram, typically specifying an agreed upon # hours / month and which model(s) will be used for the transcription. It may also specify pricing, a support contract, and/or custom models. This almost always corresponds to the Enterprise plan.

**Reply by jjmaldonis** [reply]
@ujlm that's helpful to know. I have a bunch of followup questions:

* When you say "many requests", do you have an estimate for how often that occurs? For example, are you seeing gateway timeout errors in 50% of your requests? 20%? Or maybe you've seen 30 in total (out of how many)?
* How long/large are the audio files you are sending?
* Are you sending video or audio files?
* Do you know how long the request sits before returning the gateway timeout error?
* Are you seeing any 429 rate limiting errors? How many concurrent requests are you sending?
* If there are specific request IDs that you think are relevant, please copy-paste them here :)

That's a ton of questions but this information is what allows us to make faster decisions! Thanks!

**Reply by ujlm** [reply]
@jjmaldonis Thank your for the quick reply.

### Answers to the follow-up questions:

- The occurrence of timeouts varies slightly from day to day, but it remains significant. For instance, today, 3 out of 11 requests timed out, while two days ago, 6 out of 21 requests timed out.
- On average, the audio files are about 1 hour and 30 minutes long.
- I'm sending both video and audio files for processing.
- The timeout typically happens around 20 minutes after submitting the job.
- So far, I haven't noticed any 429 rate limiting errors. If they do occur, would they be sent to the callback endpoint?
- Concurrent requests may depend on the number of jobs running simultaneously, which is usually limited to a maximum of 3 or 4 due to the lengthy processing times.
- Here are a few recent job IDs that didn't complete successfully: d66e0241-a3d8-4655-b5b9-d9a2bb148e23, 0bab8ed3-570d-4a45-8028-4fab22b6af8d, 1fedc8b4-62c4-4aaf-ae66-99414834a5f8.

### Regarding Deepgram's sales contract:

I understand that premium users receive more benefits, but as my business operates as a SaaS with a variable customer base, I chose Deepgram specifically for its flexibility in automatically scaling the transcription jobs to accommodate the changing needs through the pay-as-you-go plan. At the moment, I still have free credits, could this be the cause of the slowness and difficulties?

**Reply by jjmaldonis** [reply]
@ujlm Thanks for the additional information! Our Whisper rate limit is set to 5 concurrent requests, so it makes sense that you haven't seen 429 errors.

It's also helpful to know you are sending long audio files - Whisper is slow, and combined with a queue the requests could take longer than 20 minutes to process. We have a 20 minute timeout for Whisper requests (all other batch requests have a 10 minute timeout), so that's why you're seeing the Gateway Timeout errors.

The best way to solve this will be to break your audio files into smaller file sizes. For exapmle, if you use use 30 minute files rather than 90 minute files the processing time will be reduced significantly and you'll see less timeout errors. Also, upload speed plays a role, so if you're able to extract the audio stream from any videos then the processing time will also drop significantly.

**Reply by ujlm** [reply]
@jjmaldonis Thank you for the clarification. What's the limit rate for the other models?

I prefer to use Whisper for all languages apart from English since in my experience it performs better than Nova in these cases. The reason why I chose the Deepgram managed workflow was because of the increased speed, and the handling of large and long files. Your response and the deteriorated performance seem to suggest that the Whisper managed workflow is no longer a priority for Deepgram, is that correct?

Despite converting larger files to m4a format, I still encounter timeout errors. I'd like to avoid the complexity of splitting files, as one of the reasons I moved to Deepgram was to reduce the complexity of my workflow.

As I see it, there are two options that would solve my case:
- Either a more reliable Whisper service (which can be more expensive than the current one)
- Or an improved version of the other models so that it handles other languages than English as well as Whisper does, so there's no more reason to use Whisper in the first place

**Reply by jjmaldonis** [reply]
Hey @ujlm, I understand where you're coming from and thanks for sharing your thought process. Whisper is certainly a priority for Deepgram and supports a small but critical piece of our product offering. It may appear as if Whisper is not a priority, but the opposite is true. The deteriorated performance started a few weekends ago when Whisper usage increased drastically. The increase in usage resulted in poorer performance across the board, and our executive team, product team, and engineering team are focused on improving the performance. Within a week of the performance degradation, we improved the infrastructure around Whisper in multiple ways. Whisper may not be working as well as it was a few weeks ago when traffic was much lower, but the service can now support much more traffic. Your feedback (and everyone's!) is extremely helpful as we continue to improve Whisper.

The options you laid out to solve your use case are spot on. We are building more accurate, more cost effective, and faster models than Whisper that support other languages. For example, you may have noticed we released a Spanish version of Nova a few weeks ago. We are continuing to build and release other models with improved accuracy.

Another option that we've discussed is to split your files into multiple pieces. I understand this isn't ideal and creates a more complicated codebase on your end, and it's a temporary solution until more accurate models are available for non-English languages. But it's something actionable right now to solve the issue you're running into. We are simultaneously working on technology to solve your issue, I just can't give a public ETA on when it will be available.

The challenge is to offer a model that is: more accurate than Whisper in all languages, inexpensive, as fast as Nova, can handle large and long files, and can scale to millions of requests per day in a cost-effective way. OpenAI's version of Whisper is accurate and somewhat scalable (maximum of 50 requests/minute), but is slow and limited to small (25MB) files. Deepgram's implementation of Whisper is accurate, less expensive, can handle much larger files, but does not scale well. Nova supports English and Spanish with high accuracy, is scalable, fast, and supports large files, but does not support all languages. As you pointed out, there are tradeoffs. We would love to give you a product right now that exceeds every benchmark you're looking for - and we are working on it! It will happen, but it will take some time. I can't provide a public ETA or roadmap, but I'm excited. While that technology is being developed, you may have to make some changes on your end to support the existing global constraints of speech-to-text technology.

**Reply by ujlm** [reply]
Thank you for the explanation and the follow-up, I understand these changes may take some time. I'll make sure to keep an eye on the latest Deepgram product updates to stay up-to-date 😊

**Reply by peldszus** [reply]
Thanks @jjmaldonis for the answers & insights!

> each customer has an independent rate limit, and Whisper is rate limited to 5 concurrent requests unless you have a sales contract with us.

It would be good to show somewhere the rate limits for the different plans and models. On the pricing page the pay-as-you-go plan mentions 100 concurrent requests and lists as models Nova as well as Whisper. That's why I was mislead to conclude that I would have 100 concurrent requests for Whisper.

Besides that, I guess Whisper is the choice for most non-english/spanish applications. In my case, German is the target language. So I appreciate your effort to scale according to the increased need, or even work on Nova to support other than the two currently supported languages.

(Sorry for the late reply, just got back from vacation.)

**Reply by jjmaldonis** [reply]
Hey @peldszus, that's a good callout. We'll get the pricing page updated to reflect the Whisper rate limits.

We are working on making Nova available for more languages. If you aren't signed up for our emails, you may want to subscribe and keep an eye out for these updates!

**Comment by yousifa** [comment]
Whisper is ridiculously slow. ~45 seconds for a 1min audio file, where OpenAI processes in ~6 sec
